paper/paper.md

```diff
@@ -0,0 +1,70 @@
+
+# Statement of Need
+
+The complex gravitational fields of irregular bodies, such as asteroids and comets, are often modeled using polyhedral gravity models as they provide an analytic solution for the computation of the gravitational potential, acceleration (and second derivative) given a mesh of the body [@tsoulis2012analytical;@tsoulis2021computational]. The computation of the gravitational potential and acceleration is a computationally expensive task, especially for large meshes, which can however benefit from parallelization either over computed targets points for which we seek potential and acceleration or over the mesh. Thus, a high-performance implementation of a polyhedral gravity model is desirable. 
+
+While some research code for these models exists, they are not focused on usability and limited to FORTRAN **TODO LINK** and proprietary software like MATLAB **TODO LINK**. There is a lack of well-documented, maintained open-source implementations, particularly in modern programming languages and with a focus on scalability and performance.
+
+The presented software has already seen application in several research works. It has been used to optimize trajectories around the highly irregular comet 67P/Churyumov-Gerasimenko [@marak2023trajectory]. Further, it has been used to study the effectiveness of so-called neural density fields [@izzo2022geodesy], where it can serve as a ground truth and to pretrain neural networks [@GeodesyNetsBenchmark]. **TODO_add_more_examples**
+
+Thus, overall this model is highly versatile and can be used in a wide range of applications. We hope it will enable further research in the field, especially related to recent machine learning techniques, which typically rely on Python implementations.
+
+# Polyhedral Model
+
+On a mathematical level, the implemented model follows the approach by Petrović [@petrovic1996determination] as refined by Tsoulis and Petrović [@tsoulis2001singularities]. A comprehensive description of the mathematical foundations of the model is given in the associated student report [@jonas_report]. 
+
+Implementation-wise it makes use of the inherent parallelization opportunity of the approach as it iterates over the mesh. This parallelization is achieved via *thrust* which allows utilizing *OpenMP* and *Intel TBB*. On a finer scale, individual costly operations were investigated and, e.g., the \texttt{arctan} operations were vectorized using *xsimd*. On an application side, the user may use the implemented caching mechanism to avoid recomputation of mesh properties, such as the face normals.
```

Comment:
```suggestion
Implementation-wise, it makes use of the inherent parallelization opportunity of the approach as it iterates over the mesh. This parallelization is achieved via *thrust*, which allows utilizing *OpenMP* and *Intel TBB*. On a finer scale, individual, costly operations were investigated and, e.g., the \texttt{arctan} operations were vectorized using *xsimd*. On an application side, the user may use the implemented caching mechanism to avoid the recomputation of mesh properties, such as the face normals.
```